src/testshade/rs_simplerend.cpp

```diff
@@ -368,6 +368,18 @@ rs_trace_get(OSL::OpaqueExecContextPtr ec, OSL::ustringhash name,
 #endif
 }
 
+#ifdef __CUDA_ARCH__  // Host side uses rs_fallback implementation.
+OSL_RSOP OSL_HOSTDEVICE void*
+rs_allocate_closure(OSL::OpaqueExecContextPtr ec, size_t size, size_t alignment)
+{
+    auto sg       = (OSL::ShaderGlobals*)ec;
+    uintptr_t ptr = OIIO::round_to_multiple_of_pow2((uintptr_t)sg->renderstate,
+                                                    alignment);
+    sg->renderstate = (void*)(ptr + size);
```

Comment:
Suggest making use of the testshade/render_state.h RenderState object and add a stack_buffer data member.  Although I think RenderState is uniform accross all shades, so we might need a per thread renderstate in the ExecContext to properly handle this.

Reply:
Maybe something like

```
auto *pst = get_rs_per_shade<TestShadePerShadeState>(ec);
pst->stackbuf = (void*)(ptr + size);
```
Would require another pointer in ShaderGlobals (but hopefully we can remove some in the future).
But then a Renderer could have their own PerShadeState to implement some of this stuff.